nmostesting/suites/BCP0060101Test.py

```diff
@@ -35,110 +38,454 @@ def __init__(self, apis):
+            if not valid:
+                return test.FAIL(result)
+
+        flow_map = {flow["id"]: flow for flow in self.is04_resources["flows"]}
+        source_map = {source["id"]: source for source in self.is04_resources["sources"]}
+
+        try:
+            found_video_jxsv = False
+
+            for sender in self.is04_resources["senders"]:
+                if sender["flow_id"] is None:
+                    continue
+                if sender["flow_id"] not in flow_map:
+                    continue
+                flow = flow_map[sender["flow_id"]]
+                if flow["media_type"] != "video/jxsv":
+                    continue
+                found_video_jxsv = True
+
+                source = source_map[flow["source_id"]]
+
+                if "manifest_href" not in sender:
+                    return test.FAIL("Sender {} MUST indicate the 'manifest_href' attribute."
+                                     .format(sender["id"]))
+
+                url = sender["manifest_href"]
+                manifest_href_valid, manifest_href_response = self.do_request("GET", url)
+                if not manifest_href_valid or manifest_href_response.status_code != 200:
+                    return test.FAIL("Unexpected response from the Node API: {}"
+                                     .format(manifest_href_response))
+
+                sdp = manifest_href_response.text
+
+                payload_type = self.rtp_ptype(sdp)
+                if not payload_type:
+                    return test.FAIL("Unable to locate payload type from rtpmap in SDP file for Sender {}"
+                                     .format(sender["id"]))
+
+                found_fmtp = False
+
+                fmtp_line = "a=fmtp:{}".format(payload_type)
+                for sdp_line in sdp.split("\n"):
+                    sdp_line = sdp_line.replace("\r", "")
+
+                    if not sdp_line.startswith(fmtp_line):
+                        continue
+                    found_fmtp = True
+
+                    sdp_line = sdp_line[len(fmtp_line):]
+
+                    sdp_format_params = {}
+                    for param in sdp_line.split(";"):
+                        name, _, value = param.strip().partition("=")
+                        if name in ["interlace", "segmented"] and _:
+                            return test.FAIL("SDP '{}' for Sender {} incorrectly includes an '='"
+                                             .format(name, sender["id"]))
+                        if name in ["depth", "width", "height", "packetmode", "transmode"]:
+                            try:
+                                value = int(value)
+                            except ValueError:
+                                return test.FAIL("SDP '{}' for Sender {} is not an integer"
+                                                 .format(name, sender["id"]))
+                        sdp_format_params[name] = value
```

Comment:
Hmm, it feels like we should create/consolidate SDP parsing/validating utility functions (lots in common with parse_sdp in Node.py - although that has less explicit validation - but perhaps needs it)

Reply:
Agree, _Node.py_, _IS0502Test.py_ and here... but wanted to get to working before too much refactoring...

Reply:
How about we get this into the **bcp-006-01** branch and then think about an _SDPUtils.py_ with these things in?

Reply:
Yes, agreed!